File: experiments/bakeoff/benchmarks/python-inventory/ground_truth/reference/shop/services/customer_service.py
"""services/customer_service.py — Customer registration + lookup + address mgmt.

Backed by Store.customers. All errors are domain errors from errors.py.
"""

from ..errors import ConflictError, NotFoundError
from ..persistence import Store
from ..types.customers import Address, Customer


class CustomerService:
    def __init__(self, store: Store) -> None:
        self.store = store

    def register(self, *, id: str, name: str, email: str) -> Customer:
        if id in self.store.customers:
            raise ConflictError(f"customer with id {id} already exists")
        customer = Customer(id=id, name=name, email=email)
        self.store.customers[id] = customer
        return customer

    def get(self, id: str) -> Customer:
        customer = self.store.customers.get(id)
        if customer is None:
            raise NotFoundError(f"customer not found: {id}")
        return customer

    def add_address(self, id: str, address: Address) -> Customer:
        customer = self.get(id)
        customer.addresses.append(address)
        return customer

File: experiments/bakeoff/benchmarks/dart-inventory/ground_truth/reference/lib/services/customer_service.dart
/// services/customer_service.dart — Customer registration + lookup + address mgmt.
///
/// Backed by `Store.customers`. All errors are domain errors from `errors.dart`.

import '../errors.dart';
import '../persistence.dart';
import '../types/customers.dart';

class CustomerService {
  final Store store;
  CustomerService(this.store);

  Customer register({
    required String id,
    required String name,
    required String email,
  }) {
    if (store.customers.containsKey(id)) {
      throw ConflictError('customer with id $id already exists');
    }
    final c = Customer(id: id, name: name, email: email);
    store.customers[id] = c;
    return c;
  }

  Customer get(String id) {
    final c = store.customers[id];
    if (c == null) {
      throw NotFoundError('customer not found: $id');
    }
    return c;
  }

  Customer addAddress(String id, Address addr) {
    final c = get(id);
    c.addresses.add(addr);
    return c;
  }
}
